lightning/src/chain/channelmonitor.rs

```diff
@@ -2762,7 +2780,9 @@ impl<'a, Signer: Sign, K: KeysInterface<Signer = Signer>> ReadableArgs<&'a K>
 		let lockdown_from_offchain = Readable::read(reader)?;
 		let holder_tx_signed = Readable::read(reader)?;
 
-		read_tlv_fields!(reader, {});
+		read_tlv_fields!(reader, {
+			(1, shutdown_script, option),
```

Comment:
Hmm, I almost feel like we can skip the TLV part here, no? There's probably not a lot of reason to ever allow an empty script, so we might as well just use the original field.

Reply:
The script will be empty when a channel is created without a shutdown script, which requires using a `ChannelMonitorUpdate` at shutdown to convey it to the `ChannelMonitor`.

Reply:
Right, my point was more about shoving the `shutdown_script` in the ending TLV when its already in the main serialization body. ie do we need this line at all, or can we just use the existing read `script.is_empty { None }` check and not store a copy in the TLV at all.